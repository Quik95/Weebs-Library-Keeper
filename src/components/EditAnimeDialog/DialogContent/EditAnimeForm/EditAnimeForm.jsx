import React from "react";
import PropTypes from "prop-types";
import { useStoreState } from "easy-peasy";

// MUI
import Grid from "@material-ui/core/Grid";

import FormInput from "./FormInput/FormInput";

function EditAnimeForm({ animeId }) {
  const animeData = useStoreState(state =>
    state.animeList.list.find(({ _id }) => _id === animeId)
  );

  const {
    title,
    anilistId,
    thumbnailUrl,
    watchedEpisodes,
    latestEpisode,
    episodes
  } = animeData;

  return (
    <form noValidate autoComplete="false">
      <Grid container spacing={2}>
        <FormInput title="Title" id="title" value={title} />
        <FormInput title="Anilist ID" id="anilistId" value={anilistId} />
        <FormInput
          title="Thumbnail URL"
          id="thumbnailUrl"
          value={thumbnailUrl}
        />
        <FormInput
          title="Watched Episodes"
          id="watchedEpisodes"
          type="number"
          value={watchedEpisodes}
        />
        <FormInput
          title="Aired Episodes"
          id="airedEpisodes"
          type="number"
          value={latestEpisode}
        />
        <FormInput
          title="Total Episodes"
          id="totalEpisodes"
          type="number"
          value={episodes}
        />
      </Grid>
    </form>
  );
}

EditAnimeForm.propTypes = {
  animeId: PropTypes.string.isRequired
};

export default EditAnimeForm;
