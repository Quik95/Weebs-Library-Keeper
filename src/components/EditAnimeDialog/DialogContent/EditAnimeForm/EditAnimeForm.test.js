import EditAnimeForm from './EditAnimeForm';

describe('<EditAnimeForm/>', () => {
  const wrapper = shallow(<EditAnimeForm />);

  it('Renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
