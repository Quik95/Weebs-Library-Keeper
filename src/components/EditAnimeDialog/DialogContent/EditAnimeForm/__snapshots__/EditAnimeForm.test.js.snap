// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditAnimeForm/> Renders correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  spacing={2}
>
  <Memo(FormInput)
    id="title"
    title="Title"
  />
  <Memo(FormInput)
    id="anilistId"
    title="Anilist ID"
  />
  <Memo(FormInput)
    id="thumbnailUrl"
    title="Thumbnail URL"
  />
  <Memo(FormInput)
    id="watchedEpisodes"
    title="Watched Episodes"
    type="number"
  />
  <Memo(FormInput)
    id="airedEpisodes"
    title="Aired Episodes"
    type="number"
  />
  <Memo(FormInput)
    id="totalEpisodes"
    title="Total Episodes"
    type="number"
  />
</WithStyles(ForwardRef(Grid))>
`;
